### DISCUSSION TITLE
Sorting doesn't work for custom "Date" type.

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#287] [Q&A] [answered]
I have a custom date type that just uses `Date.now()` to generate a new date. The date itself is stored as a number therefore it should be sortable. However, it's not sorting when I specify `'asc'` or `'desc'` for the sorts. I'm using sorting in a GraphQL query, so I also cannot easily write native Mongo sorts without creating a custom endpoint.

**Comment by edobrb** [comment]
If it's stored as a number it should be working. Can you check the value on the db?

Can you show me the DataTypeAdapter for your custom Date type?

**Reply by Atlinx** [reply]
I actually don't think I have a DateTypeAdapter. Do I need one to get Dates working? So far, I've been able to write and read dates without problems.

Here's the code I have for the "scalars" setting of my EntityManager, if that has anything to do with it:
```typescript
Date: {
  generate: () => Date.now(),
}
```

**Comment by Atlinx** [comment, marked as answer]
Weird, everything seemed to sort itself out. Maybe it was an issue with my frontend that led to this.
